### ISSUE TITLE
Unable to use Dialog component with Vue.component(Buefy.Dialog.name, Buefy.Dialog)

### ISSUE THREAD

**Issue by kewlniss** [buefy/buefy#723]
Before I get into the problem. I just wanted to say thanks for making such a great library. Bulma is awesome, and the work here to make it work nicely in Vue land is spectacular. So, thank you.

### Overview of the problem
Now, I could be missing something, but I'm trying to utilize the Dialog by only loading the components I need. I've had the Select component working for a while, but trying to add the Dialog component the same way isn't working.

**Buefy** version: [0.6.4]
**Vuejs** version: [2.5.16]
**OS/Browser**: Win 64 / Chrome

### Description
Before I forget, I'm using TypeScript if that matters...

I currently am using the Select component without issue.

So I'm not using `Vue.use(Buefy) ` 
Instead, I'm using
`Vue.component(Buefy.Select.name, Buefy.Select)`

When I try to do the same thing with the Dialog
`Vue.component(Buefy.Dialog.name, Buefy.Dialog)`

I get the error that it cannot read property 'name' of undefined. It isn't finding Buefy.Dialog, like it is with Buefy.Select.

If I use `Vue.use(Buefy)` then everything works as expected.

What am I missing to use the Dialog component a la carte like I am the Select component?

**Edit:** Here is the Buefy object, showing that Dialog isn't present at all...
![image](https://user-images.githubusercontent.com/1583823/37871443-fa6112da-2fbb-11e8-9994-b89b4a011402.png)

**Comment by jtommy** [comment]
@kewlniss Did you try `import { Dialog } from 'buefy'`. Anyway `dialog` is exported as programmatic component so you can import it but use only `alert`, `confirm` and `prompt` methods.
For example: 
```
import { Dialog } from 'buefy'
Dialog.alert({
  message: 'Alert!'
})
```

**Comment by kewlniss** [comment]
@jtommy thanks for the help. 

Adding that import and then the following line:
```
Vue.prototype.$dialog = Dialog
```

to my main/boot file is what I needed. Then I was able to utilize `this.$dialog.confirm(...)` in my Single File Components.

Did I miss a list of imports from the docs?
I was going off of these pages
https://buefy.github.io/#/documentation/start
https://buefy.github.io/#/documentation/customization

which has the following information:
```
import Vue from 'vue'
import Buefy from 'buefy'
import 'buefy/lib/buefy.css'

Vue.use(Buefy)
// OR
Vue.component(Buefy.Checkbox.name, Buefy.Checkbox)
Vue.component(Buefy.Table.name, Buefy.Table)
Vue.component(Buefy.Switch.name, Buefy.Switch)
```

I should have looked at the source to see that Dialog was to be explicitly imported. Perhaps this question should be turned into an issue to update the docs?

I'll be happy to flesh out the above and add it to the docs and do a PR if that would be beneficial. I'd add it to the Import and use Buefy section here:
https://buefy.github.io/#/documentation/customization

Probably doesn't make sense to add it to the start page as I think it may confuse matters for some.

Anyway, assuming what I did was the right approach, and if you'd like me to add it to that document page (or somewhere else), just let me know.

Thanks so much for your help, and again, thanks for a great library!
